activerecord/lib/active_record/destroy_job.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class DestroyJob < ActiveJob::Base
+    queue_as { ActiveRecord::Base.queues[:destroy] }
+
+    discard_on ActiveJob::DeserializationError
+
+    def perform(record, ensuring = nil)
+      record.destroy! if eligible?(record, ensuring)
+    end
+
+    private
+      def eligible?(record, ensuring)
+        ensuring.nil? || record.public_send(ensuring)
```

Comment:
What should we do in case someone misspells `ensuring` or has renamed it later and forgotten that it was used for `destroy_later`? Too arcane to bother with, should we catch the `NoMethodError`, something else?

Reply:
I think letting the `NoMethodError` bubble up to the job runner is legit.

In the first case, where the method name in `ensuring:` is incorrect or misspelled, you’ll end up with a list of jobs to correct in your Active Job backend’s failed queue. In the second case, you’ll want to restore the method and retry the failed jobs.